simplesat/package.py

```diff
@@ -52,9 +56,33 @@ def _from_pretty_string(cls, s):
         return parser.parse_to_package(s)
 
     def __init__(self, name, version, install_requires=None):
+        """ Return a new PackageMetdata object.
+
+        Parameters
+        ----------
+        name : str
+            The name of the Python distribution, e.g. "numpy"
+        version : EnpkgVersion
+            An EnpkgVersion object describing the version of this package.
+        install_requires : tuple(tuple(str, tuple(tuple(str))))
+            A tuple of tuples mapping distribution names to disjunctions of
+            conjunctions of version constraints.
+
+            For example, a consider a package that depends on the following:
+                - nose
+                - six (> 1.2, <= 1.2.3), or >= 1.2.5-2
+                    Written as intervals, (1.2, 1.2.3] or [1.2.5-2, \infty)
+                - MKL >= 10.1, < 11
+
+            The constraint tuple representing this would be:
+                (("MKL", ((">= 10.1", "< 11"),)),
+                 ("nose", (("",),)),
+                 ("six", (("> 1.2", "<= 1.2.3"), (">= 1.2.5-2",)))
+
+        """
         self._name = name
         self._version = version
-        self._install_requires = install_requires or tuple()
+        self._install_requires = tuple(sorted(install_requires or ()))
```

Comment:
I was about to say I would rather keep it as, but we actually have to sort for determinism since we keep things in json (and json dict like python ones are not ordered), so I am fine with that